lib/readline.js

```diff
@@ -238,6 +243,24 @@ Interface.prototype.setPrompt = function(prompt) {
   this._prompt = prompt;
 };
 
+Interface.prototype.read = function() {
+  if (this._buffer.length === 0)
+    return null;
+
+  return this._buffer.shift();
+};
+
+
+Interface.prototype[Symbol.asyncIterator] = function() {
+  emitExperimentalWarning('readline Interface[Symbol.asyncIterator]');
+
+  this._enableBuffer = true;
+  this.close();
```

Comment:
We definitely should not close the readline interface when we start iterating. This will emit events like `'close'` that can be very much misleading to users.